test/json_schemer_test.rb

```diff
@@ -354,6 +354,81 @@ def test_it_returns_correct_pointers_for_ref_id
     }
   end
 
+  def test_it_returns_correct_pointers_for_a_nested_ref_id
```

Comment:
It seems like a simpler test is possible, one which doesn't use ref_resolver or multiple schema documents at all.

Reply:
I believe we need to use a ref_resolver for all refs to outside schemas regardless

Reply:
But yes a simpler test without a ref resolver would be a good addition